numba/tests/test_profile_refcount.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import division, absolute_import, print_function
+
+import numba.unittest_support as unittest
+import numpy as np
+from numba import jit
+from .support import MemoryLeakMixin, TestCase, tag
+import cProfile as profiler
+import sys
+
+
+class TestProfileRefCount(MemoryLeakMixin, unittest.TestCase):
+
+    def assert_refcount(self, arr, expect):
+        self.assertEqual(arr.base.refcount, expect)
```

Comment:
I didn't know `.refcount` and I don't think it's necessary to go through this. Just compare `sys.getrefcount(arr)` before and after the function call, IMO.